File: leetcode/ContainsDuplicate.py
from collections import defaultdict

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:


        if len(nums) == 1:
            return False

        track = defaultdict(list)

        for num in nums:
            if track.get(num):
                return True
            else:
                track[num] = 1

        return False

File: leetcode/ContainsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    
    if (nums.length == 1){
        return false;
    }
    
    const map1 = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        if(map1.get(nums[i]) != undefined) {
            return true;
        } else {
            map1.set(nums[i], 1);
        }
    }
    
    return false;
    
};
